File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    let answer = 0;
    const n = s.length;
    s = s + s;
    const ref = {
        "(" : ")",
        "{" : "}",
        "[" : "]"
    }
    
    for (let i = 0; i < n; i++) {
        const slicedS = s.slice(i, i + n);
        
        const stack = [];
        for (const item of slicedS) {
            if (!stack.length) stack.push(item);
            else {
                if (ref[stack[stack.length - 1]] === item) stack.pop();
                else stack.push(item);
            }
        }
        
        if (!stack.length) answer++
    }
    
    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
from collections import deque

def solution(s):
    answer = 0
    items = deque(list(s))
    
    
    for i in range(len(items)):
        items.rotate(-1)
        stack = []
        
        for i in range(len(items)):
            if not stack:
                stack.append(items[i])
            else:
                if items[i] == ")" and stack[-1] == "(":
                    stack.pop()
                elif items[i] == ']' and stack[-1] == '[':
                    stack.pop()
                elif items[i] == '}' and stack[-1] == '{':
                    stack.pop()
                else:
                    stack.append(items[i])
        if not stack:
            answer += 1
        

    return answer